src/dscanner/analysis/run.d

```diff
@@ -232,9 +260,15 @@ const(Module) parseModule(string fileName, ubyte[] code, RollbackAllocator* p,
 	tokens = getTokensForParser(code, config, &cache);
 	if (linesOfCode !is null)
 		(*linesOfCode) += count!(a => isLineOfCode(a.type))(tokens);
-	return dparse.parser.parseModule(tokens, fileName, p,
-		report ? toDelegate(&messageFunctionJSON) : writeMessages,
-		errorCount, warningCount);
+
+	MessageDelegate messageWriter;
+
+	if (dlgMessage !is null)
+		messageWriter = dlgMessage;
+	else
+		messageWriter = report ? toDelegate(&messageFunctionJSON) : writeMessages;
```

Comment:
I think having a new overload to parseModule with a non-optional MessageDelagate argument and no `bool report` nor `string errorFormat` argument would make the API a bit more cleaner. The MessageDelegate would then also be before the int pointers which matches the underlying API too.

The one with the old overload could then call this new overload and it would no longer create a new delegate in this method.

Reply:
fixed